### ISSUE TITLE
hundreds of compilation errors after install

### ISSUE THREAD

**Issue by SavedByZero** [thinksquirrel/fluviofx#20]
Here's a few:
```
NullReferenceException: Object reference not set to an instance of an object
UnityEditor.Scripting.ScriptCompilation.CSharpNamespaceParser.FindNamespaceForMono (System.String className, System.String source) (at /Users/builduser/buildslave/unity/build/Editor/Mono/Scripting/ScriptCompilation/CSharpNamespaceParser.cs:84)
UnityEditor.Scripting.ScriptCompilation.CSharpNamespaceParser.GetNamespace (System.String sourceCode, System.String className, System.String[] defines) (at /Users/builduser/buildslave/unity/build/Editor/Mono/Scripting/ScriptCompilation/CSharpNamespaceParser.cs:49)
Rethrow as IllegalNamespaceParsing: Searching for classname: 'FluvioFXCollisionUtility' caused error in CSharpNameParser
UnityEditor.Scripting.ScriptCompilation.CSharpNamespaceParser.GetNamespace (System.String sourceCode, System.String className, System.String[] defines) (at /Users/builduser/buildslave/unity/build/Editor/Mono/Scripting/ScriptCompilation/CSharpNamespaceParser.cs:53)
UnityEditor.Scripting.Compilers.CSharpLanguage.GetNamespaceNewRuntime (System.String filePath, System.String definedSymbols, System.String[] defines) (at /Users/builduser/buildslave/unity/build/Editor/Mono/Scripting/Compilers/CSharpLanguage.cs:90)
UnityEditor.Scripting.Compilers.CSharpLanguage.GetNamespace (System.String filePath, System.String definedSymbols) (at /Users/builduser/buildslave/unity/build/Editor/Mono/Scripting/Compilers/CSharpLanguage.cs:133)
UnityEditor.Scripting.ScriptCompilers.GetNamespace (System.String file, System.String definedSymbols) (at /Users/builduser/buildslave/unity/build/Editor/Mono/Scripting/ScriptCompilers.cs:109)
Library/PackageCache/com.thinksquirrel.fluviofx@99cca250b4e55486e22785c8eb0e522bf0151734/Editor/Blocks/Collision/CollisionAABox.cs(9,50): error CS0122: 'CollisionAABox' is inaccessible due to its protection level
Library/PackageCache/com.thinksquirrel.fluviofx@99cca250b4e55486e22785c8eb0e522bf0151734/Editor/Blocks/Collision/CollisionPlane.cs(9,50): error CS0122: 'CollisionPlane' is inaccessible due to its protection level
Library/PackageCache/com.thinksquirrel.fluviofx@99cca250b4e55486e22785c8eb0e522bf0151734/Editor/Blocks/Solver/FluvioFXBlock.cs(12,36): error CS0122: 'VFXBlock' is inaccessible due to its protection level

```
I have hundreds of these.  I've tried installing it on 2019.2 or later and it gives me other errors about some class that's been reformatted into a package for 2019.  I tried again in Unity 2018.4.2 and I get these errors.  Following the readme's advice did not work -- it actually gets worse when I remove that entry from the package.json file, let it uninstall, and then put it back in.

**Comment by lmontoute** [comment]
Thanks - have not tried 2019.2. Will look into it this weekend.

**Comment by SavedByZero** [comment]
2018.4.2f1 gives me those errors as well.  
They all go away when I remove this line from my manifest:
    "com.thinksquirrel.fluviofx": "https://github.com/thinksquirrel/fluviofx.git",

All of them are "inaccessible due to protection level" errors in the Fluvio code files.  Hope that helps.

**Comment by lmontoute** [comment]
Understood.

This is likely due to an update in HDRP/VFX breaking FluvioFX. To be expected - we patch the files for VFX system for now (we've been talking with Unity a bit to streamline the process for VFX extensions so that this won't be needed in the future). So we just need to update the FluvioFX installer to match the updates in those packages.

**Comment by lmontoute** [comment]
Hi @SavedByZero,

Can you give the latest pull request a try and see if that fixes your issue? Note the change with installation instructions as well. You may need to remove and reinstall the needed packages in order to get things working.

**Comment by lmontoute** [comment]
Fixed by #23

**Comment by SavedByZero** [comment]
Sorry for the late reply, this has been a back burner research project for me and I got swamped. Anyhow, yes, I can install it now, thanks. Trying to use it afterwards crashes my little MacBook but I should have expected that (I can move the project to my desktop PC). 

Looking forward to any examples of how to use it in the future — specifically, I am interested in making a water-through-pipes style puzzle game in the spirit of Pipe Dream.

**Comment by lmontoute** [comment]
Hm, a macOS crash is actually new to me, do you have problems with the VFX system in general on that MacBook? Personally, my MacBook gets lower performance (as expected) but otherwise runs fine.

This may not be a FluvioFX bug, but I highly recommend reporting the issue to Unity. They've communicated that even though FluvioFX uses experimental features, they can take a look at projects with it that experience any issues like these.

Also: awesome project! I'm personally a big fan of fluid puzzles, so let me know if you make something neat 🙂

**Comment by tumml3r** [comment]
I'm having the same issue-- upon installing, I get 230+ errors, mostly 'no suitable method' and 'inaccessible due to protection level'. Supposedly this was fixed with #23 so I'm confused why I'd be having this issue now. Any ideas?
